### PR TITLE
Remove dependency on `com.google.gdata.data/gdata-core` (connect #2658)

### PR DESCRIPTION
**Pull request by iperdomo**
* We had an implicit dependency on commons-lang/commons-lang
  via org.apache.velocity/velocity, we just made it explicit
* We're using javax.xml.bind.DatatypeConverter for Base64 encoding
  but we need to move to Base64 API on Java 8 when possible
* Fixed the @Nullable annotation to use `javax.annotation.*` and
  not a library one
* We're not using TokenUtility class
* Related #555 

## Checklist
* [x] Connect the issue
* [ ] Test plan
* [x] Copyright header
* [x] Code formatting
* [ ] Documentation

### CODE REVIEW

File: GAE/src/com/gallatinsystems/common/util/MD5Util.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- *  Copyright (C) 2010-2012 Stichting Akvo (Akvo Foundation)
+ *  Copyright (C) 2010-2018 Stichting Akvo (Akvo Foundation)
```
Comment (muloem) [inline GAE/src/com/gallatinsystems/common/util/MD5Util.java]: Small note: following the strict convention this would be `2010-2012,2028`

Comment (iperdomo) [reply GAE/src/com/gallatinsystems/common/util/MD5Util.java]: are we following the _strict convention_?, if so, we need to fix *lots* of copyright headers

**Comment by dlebrero** [comment]
The pom has a reference to the eib.ac.uk. Could you please remove it and test that we do not have any other dependency against that repo? Maybe by removing your own local maven repo (or moving it temporarily to another path)

**Comment by iperdomo** [comment]
Tested by _clearing_ the cache. Locally works as expected.

````diff
diff --git a/docker-compose.yml b/docker-compose.yml
index 74d86a718..16a5e9e2c 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -16,5 +16,5 @@ services:
      - 5005:5005
     volumes:
      - .:/app/src/:delegated
-     - ~/.m2:/home/akvo/.m2:delegated
-     - ~/.m2:/root/.m2:delegated
+     - /tmp/m2:/home/akvo/.m2:delegated
+     - /tmp/m2:/root/.m2:delegated
````

File: GAE/src/com/gallatinsystems/common/util/MD5Util.java (hunk 1 above)
Comment (iperdomo) [reply GAE/src/com/gallatinsystems/common/util/MD5Util.java]: e.g. according to `git log` this file was modified in 2011,2012,2014,2018

     git log --date=short GAE/src/com/gallatinsystems/common/util/MD5Util.java | awk '/^Date:/ {print $2}'

I guess that something similar applies to several files

Comment (muloem) [reply GAE/src/com/gallatinsystems/common/util/MD5Util.java]: This of course applies to all files but following the boy scout rule, this is the file you were working on at the moment and can actually do something about without disrupting too much else.

**Pull request merged**